server/storage/mvcc/key_index_test.go

```diff
@@ -462,9 +489,14 @@ func TestKeyIndexCompactAndKeep(t *testing.T) {
 		if !reflect.DeepEqual(ki, kiclone) {
 			t.Errorf("#%d: ki = %+v, want %+v", i, ki, kiclone)
 		}
-		if !reflect.DeepEqual(am, tt.wam) {
-			t.Errorf("#%d: am = %+v, want %+v", i, am, tt.wam)
+
+		if _, ok := tombstones[tt.compact]; ok {
+			assert.Equal(t, len(am), 0, "#%d: ki = %d, keep result wants empty because tombstone", i, ki)
```

Comment:
Also as mentioned above, you should be able to programmatically check whether `tt.compact` is a tombstone
```suggestion
			assert.Equal(t, 0, len(am), "#%d: ki = %d, keep result wants empty because tombstone", i, ki)
```

Reply:
Updated.